src/RunspaceProviders/Kubernetes/VMware.ScriptRuntimeService.K8sRunspaceProvider/K8sRunspaceProvider.cs

```diff
@@ -615,6 +561,202 @@ public class K8sRunspaceProvider : IRunspaceProvider {
+                     eventList = _client.CoreV1.ListNamespacedEvent("ingress-nginx");
+                  } catch (Exception exc) {
+                     LogException(exc);
+                     result = new K8sWebConsoleInfo {
+                        Id = result.Id,
+                        CreationState = RunspaceCreationState.Error,
+                        CreationError = new RunspaceProviderException(
+                           string.Format(
+                              Resources.K8sRunspaceProvider_WaitCreateComplation_ListEvents, result.Id),
+                           exc)
+                     };
+                  }
+
+                  if (eventList?.Items.Any(i => IsNginxReloadEventAfter(i, creationTime)) ?? false) {
+                     var reloadEvent = eventList.Items.First(i => IsNginxReloadEventAfter(i, creationTime));
+                     _logger.LogDebug($"NGINX reload event found {reloadEvent}");
+                     break;
+                  }
+
+                  Thread.Sleep(retryIntervalMs);
+
+                  retryCount++;
+               } while (retryCount < maxRetryCount);
+
+               if (retryCount >= maxRetryCount) {
+                  // Timeout
+                  result = new K8sWebConsoleInfo {
+                     Id = result.Id,
+                     CreationState = RunspaceCreationState.Error,
+                     CreationError = new RunspaceProviderException(Resources.K8sRunspaceProvider_WaitCreateComplition_TimeOut)
+                  };
+               } else {
+                  // Success, everything should be in place
+                  result = new K8sWebConsoleInfo {
+                     Id = result.Id,
+                     CreationState = RunspaceCreationState.Ready
+                  };
+               }
+            }
+         }
+
+         return result;
+      }
+
+      private static bool IsNginxReloadEventAfter(Corev1Event e, DateTime since) {
+         return e.LastTimestamp?.CompareTo(since) > 0 &&
+            e.Type.Equals("Normal", StringComparison.InvariantCultureIgnoreCase) &&
+            e.Reason.Equals("RELOAD", StringComparison.InvariantCultureIgnoreCase) &&
+            e.Message.Equals("NGINX reload triggered due to a change in configuration");
+      }
+
+      private V1Pod WaitForPodCreation(string name) {
+         V1Pod pod = null;
+         try {
+            _logger.LogDebug($"Waiting k8s Pod '{name}' to become ready");
+            _logger.LogDebug($"K8s API Call ReadNamespacedPod: {name}");
+            pod = _client.CoreV1.ReadNamespacedPod(name, _namespace);
+         } catch (Exception exc) {
+            LogException(exc);
+            throw new RunspaceProviderException(
+                  string.Format(
+                     Resources.K8sRunspaceProvider_WaitCreateComplation_PodNotFound, name),
+                  exc);
+         }
+
+         if (pod != null) {
```

Comment:
When is pod null and why not handle the null case instead?

Reply:
fixed